lib/workers/bin/heartbeat_check.rb

```diff
@@ -0,0 +1,50 @@
+require "optparse"
+
+options = {}
+opt_parser = OptionParser.new do |opts|
+  opts.banner = "usage: #{File.basename $PROGRAM_NAME, '.rb'} [HEARTBEAT_FILE]"
+
+  opts.on("-b=HBFILE", "--heartbeat-file=HBFILE", "Heartbeat file to read (overrides arg val)") do |val|
+    options[:heartbeat_file] = val
+  end
+
+  opts.on("-g=GUID", "--guid=GUID", "Use this guid for finding the heartbeat file") do |val|
+    options[:guid] = val
+  end
+
+  opts.on("-v", "--[no-]verbose", "Verbose output") do |val|
+    options[:verbose] = val
+  end
+
+  opts.on("-h", "--help", "Displays this help") do
+    puts opts
+    exit
+  end
+end
+opt_parser.parse!
+
+require File.expand_path("../../miq_defaults.rb", __FILE__)
+heartbeat_file   = options[:heartbeat_file] || ARGV[0]
+heartbeat_file ||= Workers::MiqDefaults.heartbeat_file(options[:guid])
+
+# Determine if we are within the timeout, otherwise exit 1
+if File.exist?(heartbeat_file)
+  require "time"
+  current_time = Time.now.utc
+
+  mtime    = File.mtime(heartbeat_file).utc
+  contents = File.read(heartbeat_file)
+  timeout  = if contents.empty?
+               (mtime + Workers::MiqDefaults.heartbeat_timeout).utc
+             else
+               Time.parse(contents).utc
```

Comment:
So, this is the "more accurate" way of determining the timeout, but I guess I failed to mention this in my commit messages and the PR description.  I will try and explain:

So, if you look where `Workers::MiqDefaults.heartbeat_timeout` is actually called from, you will see if is called from here:

https://github.com/ManageIQ/manageiq/blob/3ea55fa/app/models/miq_server/worker_management/monitor/validation.rb#L4-L19

This method takes a hearbeat timeout value from the settings (or defaults it to `Workers::MiqDefaults.heartbeat_timeout` if that does not exist), and then compares it against the `w.last_heartbeat` value.  Action is then taken if the heartbeat is not responding, but it is based on the `w.last_heartbeat` and the timeout value in the settings.

To avoid having to do a DB lookup in this script, I put the calculation burden of determining when this timeout time was on the worker writing the file, avoiding the heartbeat script having to deal with that.  Then, the only thing that the script needs to do is read that value and determine if that time has already passed or not.

The `mtime` is now just as a fallback incase the worker didn't write to the file properly, and needs to make a "best guess".  That said, we maybe should do a `rescue` and use the `mtime` and not concern ourselves with checking the `contents.empty?`, or just let the script fail.